src/main/java/io/vertx/core/json/jackson/JsonArrayDeserializer.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2011-2023 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.core.json.jackson;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import io.vertx.core.json.JsonArray;
+
+import java.io.IOException;
+import java.util.List;
+
+class JsonArrayDeserializer extends JsonDeserializer<JsonArray> {
+    @Override
+    public JsonArray deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+        return new JsonArray(p.<List<Object>>readValueAs(new TypeReference<>() {
```

Comment:
Given this is going to be invoked a lot, in most applications, I think we can spare `TypeReference` object creations by using a constant.

Reply:
I created a single constant to share between both of them. If you meant to create individual instances I can do that as well